portfolio/src/main/webapp/script.js

```diff
@@ -34,3 +34,62 @@ function addRandomFact() {
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact; 
 }
+
+const size = 30;
+var canvas;
+var ctx;
+var x = new Array(5);
+var y = new Array(5);
+
+function initSnake() {
+  canvas = document.getElementById('snakeCanvas'); // Get the canvas element by Id
+  ctx = canvas.getContext('2d'); // Canvas 2d rendering context
+  for (var z=0; z < 5; z++) {
+    x[z] = 10;
+    y[z] = 10;
+  }
+  drawSnake();
+}
+	
+function drawSnake() {
+  ctx.clearRect(0, 0, canvas.width, canvas.height);
+  
+  for (var z=0; z<5; z++) {
+    ctx.beginPath();
+    ctx.rect(x[z], y[z], size, size);
+    ctx.fillStyle = "#FF0000";
+    ctx.fill();
+    ctx.closePath();
+  }
+}
+
+const LEFT_KEY = 37;
+const RIGHT_KEY = 39;
+const UP_KEY = 38;
+const DOWN_KEY = 40;
+
+//move snake inside the canvas using arrow keys
+window.onkeydown = function(event) {
+  for (var z = 4; z > 0; z--) {
+    x[z] = x[(z - 1)];
+    y[z] = y[(z - 1)];
+  }
+  
+  var keyPr = event.keyCode; //Key code of key pressed
+  
+  if(keyPr == RIGHT_KEY && x[0]<=canvas.width - 40) { 
```

Comment:
What is the intended range for valid values? -10 to (width - 20)?

Reply:
The intended range for width is from 10 to (max width - 40). The range for height is from 10 to (max height - 40).